src/script/api/script_base.cpp

```diff
@@ -10,36 +10,46 @@
 #include "../../stdafx.h"
 #include "script_base.hpp"
 #include "script_error.hpp"
+#include "script_log.hpp"
 
 #include "../../safeguards.h"
 
-/* static */ uint32 ScriptBase::Rand()
+/* static */ SQInteger ScriptBase::Rand()
 {
 	return ScriptObject::GetRandomizer().Next();
 }
 
-/* static */ uint32 ScriptBase::RandItem(int unused_param)
+/* static */ SQInteger ScriptBase::RandItem(SQInteger unused_param)
 {
 	return ScriptBase::Rand();
 }
 
-/* static */ uint ScriptBase::RandRange(uint max)
+/* static */ SQInteger ScriptBase::RandRange(SQInteger max)
 {
+	if (max < 0 || max > UINT32_MAX) {
+		ScriptLog::Warning("Clamping RandRange() parameter to 0..MAX(uint32)");
```

Comment:
ScriptLog::Warning is a bit too intrusive, better just tell a clamping may happen in the description, rather than outputing it into the log.

Reply:
Agreed, will update.